units/src/amount/signed.rs

```diff
@@ -126,7 +142,9 @@ impl SignedAmount {
     ///
     /// If the amount is too big (positive or negative) or too precise.
     ///
-    /// Please be aware of the risk of using floating-point numbers.
+    /// **Warning:** due to precision loss, using floats for financial operations is generally not
+    /// recommended. Try to avoid it by always using integer number of satoshis or string-encoded
+    /// btc in APIs that require it.
```

Comment:
As an Australian reading this it sounds a bit patronising. Perhaps @apoelstra can weigh in on how an American would interpret this (since he is closer to them than I am).

Perhaps this is better:

 ```suggestion
    /// recommended. It can be avoided by using an integer number of satoshis or string-encoded
    /// btc in APIs that require it.
```

Reply:
Yeah, I like your wording better. I hoped the "commanding" tone will be dampened by the previous word "recommended" and couldn't think of a better way to say it.